streams/src/main/java/org/apache/kafka/streams/processor/internals/TaskManager.java

```diff
@@ -81,9 +81,8 @@
                 final UUID processId,
                 final String logPrefix,
                 final StreamsMetricsImpl streamsMetrics,
-                final StreamThread.AbstractTaskCreator<? extends Task> activeTaskCreator,
-                final StreamThread.AbstractTaskCreator<? extends Task> standbyTaskCreator,
-                final Map<TaskId, Producer<byte[], byte[]>> taskProducers,
+                final ActiveTaskCreator activeTaskCreator,
```

Comment:
In `shutdown(final boolean clean)` we should also release task producers as well right?

Reply:
Yep. Good catch.